ledger/metrics/src/main/scala/com/daml/metrics/MetricsReporter.scala

```diff
@@ -94,12 +93,12 @@ object MetricsReporter {
       new URI(value)
     } catch {
       case NonFatal(exception) =>
-        throw new InvalidConfigException(cliHint + " " + exception.getMessage)
+        throw new RuntimeException(cliHint + " " + exception.getMessage)
     }
 
   val cliHint: String =
     """Must be one of "console", "csv:///PATH", "graphite://HOST[:PORT][/METRIC_PREFIX]", or "prometheus://HOST[:PORT]"."""
 
-  private def invalidRead: InvalidConfigException =
-    new InvalidConfigException(cliHint)
+  private def invalidRead =
+    new RuntimeException(cliHint)
```

Comment:
I guess you had to do this to avoid grabbing a dependency. Maybe have a look if it could make sense to move `InvalidConfigException` here.

Reply:
I already looked into that and adding a whole exception hierarchy into the metrics project did not seem right.

Reply:
Cool, then please don't drop the cause (it made sense to drop it originally because `InvalidConfigException` extends `NoStackTrace`). Alternatively you can simply not re-throw.

Reply:
OK, I propagated the exception into the RuntimeException with the existing msg. I hope this is sufficient